beacon_node/network/src/sync/range_sync/range.rs

```diff
@@ -370,8 +370,9 @@ impl<T: BeaconChainTypes> RangeSync<T> {
     /// for this peer. If so we mark the batch as failed. The batch may then hit it's maximum
     /// retries. In this case, we need to remove the chain and re-status all the peers.
     fn remove_peer(&mut self, network: &mut SyncNetworkContext<T::EthSpec>, peer_id: &PeerId) {
-        if let Some((index, ProcessingResult::RemoveChain)) =
-            self.chains.head_finalized_request(|chain| {
+        while let Some((index, ProcessingResult::RemoveChain)) = self
```

Comment:
Won't this end early if there are three chains with the peer. 
The first chain, returns ProcessingResult::KeepChain. Then this loop will end and the peers in the next two chains will remain there?

Reply:
Yep, you are right. Will fix it.